diff -P -u -x '*~' -x '\.*' -r SF35-Thingomatic-HBP/alterations/end.gcode SF35-Thingomatic-ABP/alterations/end.gcode
--- SF35-Thingomatic-HBP/alterations/end.gcode	2011-01-26 22:52:40.000000000 +0100
+++ SF35-Thingomatic-ABP/alterations/end.gcode	2011-01-26 22:52:40.000000000 +0100
@@ -15,7 +15,9 @@
 M18 (Turn off steppers)
 (**** begin eject ****)
 M6 T0 (wait for toolhead parts (nozzle, HBP, etc) to reach temperature)
-M01 (Remove the object then click yes.)
+M106 (conveyor on)
+G04 P14000 (wait t/1000 seconds)
+M107 (conveyor off)
 (**** end eject ****)
 (**** begin cool for safety ****)
 M104 S0 T0 (set extruder temperature)
