diff -P -u -x '*~' -x '\.*' -r SF35-cupcake-HBP/alterations/end.gcode SF35-cupcake-ABP/alterations/end.gcode
--- SF35-cupcake-HBP/alterations/end.gcode	2011-02-03 03:55:19.000000000 +0100
+++ SF35-cupcake-ABP/alterations/end.gcode	2011-02-03 03:58:08.000000000 +0100
@@ -1,12 +1,23 @@
 (**** Beginning of end.gcode ****)
-M104 S0 T0 (set extruder temperature)
-M109 S0 T0 (set heated-build-platform temperature)
+M109 S95 T0 (set heated-build-platform temperature)
+G1 X0 Y54 F3300.0 (move platform to ejection position)
 M102 (Extruder on, reverse)
 G04 P2000 (Wait t/1000 seconds)
 M103 (Extruder off)
-
-(**** begin move to cooling position ****)
-G91
-G1 Z10
-(**** end move to cooling position ****)
+G04 P90000 (wait t/1000 seconds)
+M106 (conveyor on)
+G04 P2000 (wait t/1000 seconds)
+M104 S225 T0 (set extruder temperature)
+M109 S130 T0 (set heated-build-platform temperature)
+G04 P7000 (wait t/1000 seconds)
+M107 (conveyor off)
+(**** start wipe ****)
+G1 X-54 Y-15.5 Z6.0 F2500.0
+G1 X-54 Y15.5 Z6.0 F2500.0
+(**** end wipe ****)
+G1 X0 Y0 F3300.0 (move nozzle to center)
+G1 X0 Y0 Z0 F3300.0 (move nozzle to origin)
+M104 S0 T0 (set extruder temperature)
+M109 S0 T0 (set heated-build-platform temperature)
 (**** end of end.gcode ****)
+
diff -P -u -x '*~' -x '\.*' -r SF35-cupcake-HBP/alterations/start.gcode SF35-cupcake-ABP/alterations/start.gcode
--- SF35-cupcake-HBP/alterations/start.gcode	2011-02-03 03:55:19.000000000 +0100
+++ SF35-cupcake-ABP/alterations/start.gcode	2011-02-03 03:58:08.000000000 +0100
@@ -3,18 +3,21 @@
 (**** begin initilization commands ****)
 G21 (set units to mm)
 G90 (set positioning to absolute)
-M108 S255 (set extruder speed to maximum)
-M103 (Make sure extruder is off)
-M104 S225 T0 (set extruder temperature)
-M109 S125 T0 (set heated-build-platform temperature)
 G92 X0 Y0 Z0 (set origin to current position)
-(**** end initialization commands ****)
-G0 Z15 (Move up for test extrusion)
-M108 S255 (Extruder speed = max)
-M6 T0 (Wait for tool to heat up)
+M108 S255 (set extruder speed to maximum)
+M104 S220 T0 (set extruder temperature)
+M109 S120 T0 (set heated-build-platform temperature)
+(**** end initilization commands ****)
+(**** begin pre-wipe commands ****)
+G1 X-54 Y-30 Z6 F3300.0 (move to waiting position)
+M6 T0 (wait for tool to heat up)
+G04 P85000 (Wait t/1000 seconds)
 M101 (Extruder on, forward)
-G04 P5000 (Wait 5 seconds)
+G04 P6500 (Wait t/1000 seconds)
 M103 (Extruder off)
-M01 (The heater is warming up and will do a test extrusion.  Click yes after you have cleared the nozzle of the extrusion.)
-G0 Z0(Go back to zero.)
-(**** end of start.gcode ****)
+(**** end pre-wipe commands ****)
+G1 X-30 Y30 Z6 F3300.0 (move to initial position)
+G1 Z0 F3300.0 (Go back down)
+M101 (start extruder, fwd)
+G1 X0 Y0 Z0 F2400.0 (move to origin)
+(**** end of start.gxode ****)
diff -P -u -x '*~' -x '\.*' -r SF35-cupcake-HBP/profiles/extrusion/ABS/wipe.csv SF35-cupcake-ABP/profiles/extrusion/ABS/wipe.csv
--- SF35-cupcake-HBP/profiles/extrusion/ABS/wipe.csv	2011-02-03 03:55:19.000000000 +0100
+++ SF35-cupcake-ABP/profiles/extrusion/ABS/wipe.csv	2011-02-03 03:55:19.000000000 +0100
@@ -2,14 +2,14 @@
 Name                          	Value
 WindowPosition	700+0
 Open File for Wipe	
-Activate Wipe	False
-Location Arrival X (mm):	52.0
-Location Arrival Y (mm):	-57.0
-Location Arrival Z (mm):	10.0
-Location Departure X (mm):	52.0
-Location Departure Y (mm):	-30.0
-Location Departure Z (mm):	10.0
-Location Wipe X (mm):	52.0
-Location Wipe Y (mm):	-55.0
-Location Wipe Z (mm):	10.0
+Activate Wipe	True
+Location Arrival X (mm):	-54.0
+Location Arrival Y (mm):	-30.0
+Location Arrival Z (mm):	6.0
+Location Departure X (mm):	-54.0
+Location Departure Y (mm):	4.0
+Location Departure Z (mm):	6.0
+Location Wipe X (mm):	-54.0
+Location Wipe Y (mm):	0.0
+Location Wipe Z (mm):	6.0
 Wipe Period (layers):	99999999
