diff -P -u -x '*~' -x '\.*' -r SF35-Makerbot-baseline/alterations/end.gcode SF35-cupcake-baseline/alterations/end.gcode
--- SF35-Makerbot-baseline/alterations/end.gcode	2011-02-03 03:54:29.000000000 +0100
+++ SF35-cupcake-baseline/alterations/end.gcode	2011-02-03 03:54:28.000000000 +0100
@@ -1,2 +1,12 @@
 (**** Beginning of end.gcode ****)
+M104 S0 T0 (set extruder temperature)
+M109 S0 T0 (set heated-build-platform temperature)
+M102 (Extruder on, reverse)
+G04 P2000 (Wait t/1000 seconds)
+M103 (Extruder off)
+
+(**** begin move to cooling position ****)
+G91
+G1 Z10
+(**** end move to cooling position ****)
 (**** end of end.gcode ****)
diff -P -u -x '*~' -x '\.*' -r SF35-Makerbot-baseline/alterations/start.gcode SF35-cupcake-baseline/alterations/start.gcode
--- SF35-Makerbot-baseline/alterations/start.gcode	2011-02-03 03:52:18.000000000 +0100
+++ SF35-cupcake-baseline/alterations/start.gcode	2011-02-03 03:52:11.000000000 +0100
@@ -1,4 +1,5 @@
 (**** beginning of start.gcode ****)
+(This file is for a MakerBot Cupcake CNC)
 (**** begin initialization commands ****)
 G21 (set units to mm)
 G90 (set positioning to absolute)
@@ -6,5 +7,14 @@
 M103 (Make sure extruder is off)
 M104 S225 T0 (set extruder temperature)
 M109 S125 T0 (set heated-build-platform temperature)
+G92 X0 Y0 Z0 (set origin to current position)
 (**** end initialization commands ****)
+G0 Z15 (Move up for test extrusion)
+M108 S255 (Extruder speed = max)
+M6 T0 (Wait for tool to heat up)
+M101 (Extruder on, forward)
+G04 P5000 (Wait 5 seconds)
+M103 (Extruder off)
+M01 (The heater is warming up and will do a test extrusion.  Click yes after you have cleared the nozzle of the extrusion.)
+G0 Z0(Go back to zero.)
 (**** end of start.gcode ****)
